code/datums/components/mob_harvest.dm

```diff
@@ -52,26 +56,36 @@
 	if(var_name == NAMEOF(src, amount_ready) && var_value != amount_ready)
 		amount_changed = TRUE
 	. = ..()
-	if(amount_changed)
-		SEND_SIGNAL(parent, COMSIG_LIVING_HARVEST_UPDATE, amount_ready)
+	if(amount_changed && !iscarbon(parent))
+		var/mob/living/living_parent = parent
+		living_parent.update_appearance(UPDATE_ICON_STATE)
 
 /datum/component/mob_harvest/process(delta_time)
 	///only track time if we aren't dead and have room for more items
 	var/mob/living/harvest_mob = parent
-	if(harvest_mob.stat != DEAD && amount_ready < max_ready)
-		item_generation_time -= delta_time
-		if(item_generation_time <= 0)
-			item_generation_time = item_generation_wait
-			amount_ready++
-			SEND_SIGNAL(parent, COMSIG_LIVING_HARVEST_UPDATE, amount_ready)
+	if(harvest_mob.stat == DEAD || amount_ready >= max_ready)
+		return
+
+	item_generation_time -= delta_time
+	if(item_generation_time > 0)
+		return
+
+	item_generation_time = item_generation_wait
+	amount_ready++
+	if(iscarbon(parent))
+		return
+
+	var/mob/living/living_parent = parent
+	living_parent.update_appearance(UPDATE_ICON_STATE)
 
 /datum/component/mob_harvest/RegisterWithParent()
 	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
 	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/on_attackby)
+	if(!iscarbon(parent)) // Only do update_icon_state business on non-carbon mobs
+		RegisterSignal(parent, COMSIG_ATOM_UPDATE_ICON_STATE, .proc/on_update_icon_state)
```

Comment:
may as well have it update_appearance automatically here too instead of during sheep/init since that was only to sync up the sheep's icon with the component's